apps/rush-lib/src/logic/ProjectTaskSelector.ts

```diff
@@ -33,18 +37,15 @@ export interface IProjectTaskSelectorOptions {
  */
 export class ProjectTaskSelector {
   private readonly _options: IProjectTaskSelectorOptions;
-  private readonly _projectChangeAnalyzer: ProjectChangeAnalyzer;
-  private readonly _phasesToRun: Iterable<string>;
-  private readonly _phases: Map<string, IPhase>;
-  private readonly _customParametersByPhaseName: Map<string, string[]>;
+  private readonly _phasesToRun: Set<IPhase>;
+  private readonly _phases: IPhase[];
+  private readonly _customParametersByPhase: Map<IPhase, string[]>;
 
   public constructor(options: IProjectTaskSelectorOptions) {
     this._options = options;
-    this._projectChangeAnalyzer =
-      options.projectChangeAnalyzer || new ProjectChangeAnalyzer(options.rushConfiguration);
-    this._phasesToRun = options.phasesToRun;
-    this._phases = options.phases;
-    this._customParametersByPhaseName = new Map();
+    this._phasesToRun = new Set(options.phasesToRun);
+    this._phases = Array.from(options.phases.values());
```

Comment:
How do we know the values in a map are going to be in a particular order?

Reply:
The ECMAScript Map specification says that the iterator enumerates entries in insertion order.

Reply:
It'd be safer with future refactors to just record the phases into an ordered array in the first place and pass that in.